src/services/io/podio/CMakeLists.txt

```
@@ -31,21 +27,23 @@ plugin_glob_all(${PLUGIN_NAME})
 
 # Generate the datamodel header files.
 add_custom_command(
-        OUTPUT ${podio_BINARY_DIR}/datamodel_glue.h ${podio_BINARY_DIR}/datamodel_includes.h
-        COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py WORKING_DIR=${podio_BINARY_DIR} EDM4HEP_INCLUDE_DIR=${EDM4HEP_INCLUDE_DIR} EDM4EIC_INCLUDE_DIR=${EDM4EIC_INCLUDE_DIR}
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
+        COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+                WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
+                EDM4HEP_INCLUDE_DIR=$<REMOVE_DUPLICATES:$<TARGET_PROPERTY:EDM4HEP::edm4hep,INTERFACE_INCLUDE_DIRECTORIES>>
+                EDM4EIC_INCLUDE_DIR=$<REMOVE_DUPLICATES:$<TARGET_PROPERTY:EDM4EIC::edm4eic,INTERFACE_INCLUDE_DIRECTORIES>>
         MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
 )
 
 # Add include directories (works same as target_include_directories)
-message(STATUS "fmt_INCLUDE_DIR=${fmt_INCLUDE_DIR}")
-plugin_include_directories(${PLUGIN_NAME} SYSTEM PUBLIC ${INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_BINARY_DIR}/src ${CMAKE_CURRENT_SOURCE_DIR} ${EICRECON_SOURCE_DIR}/src)
+plugin_include_directories(${PLUGIN_NAME} PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
 
 # Add libraries (works same as target_include_directories)
-plugin_link_libraries(${PLUGIN_NAME} EDM4HEP::edm4hep EDM4HEP::edm4hepDict EDM4EIC::edm4eic EDM4EIC::edm4eic_utils podio::podioRootIO)
+plugin_link_libraries(${PLUGIN_NAME} fmt::fmt EDM4HEP::edm4hep EDM4HEP::edm4hepDict EDM4EIC::edm4eic EDM4EIC::edm4eic_utils podio::podioRootIO)
 
 # Create a ROOT dictionary with the vector<edm4hep::XXXData> types defined. Without
 # this, root will complain about not having a compiled CollectionProxy.
-root_generate_dictionary(G__datamodel_vectors ${podio_BINARY_DIR}/datamodel_includes.h MODULE ${PLUGIN_NAME}_plugin LINKDEF datamodel_LinkDef.h)
+root_generate_dictionary(G__datamodel_vectors ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h MODULE ${PLUGIN_NAME}_plugin LINKDEF datamodel_LinkDef.h)
```

```suggestion
root_generate_dictionary(G__datamodel_vectors ${datamodel_BINARY_DIR}/datamodel_includes.h MODULE ${PLUGIN_NAME}_plugin LINKDEF datamodel_LinkDef.h)
```
Make things searchable.

---

I'm not sure about this one. It also means that the internals of the project would depend on the name of the project. If someone changes the name from `datamodel` to something else, wouldn't they reasonably expect that only external users would need updating?